sw/device/lib/crypto/drivers/kmac.c

```diff
@@ -13,20 +13,242 @@
+
+  // Configure max for entropy period
+  uint32_t entropy_period = KMAC_ENTROPY_PERIOD_REG_RESVAL;
+  entropy_period = bitfield_field32_write(
+      entropy_period, KMAC_ENTROPY_PERIOD_PRESCALER_FIELD, kKmacPrescalerMax);
+  entropy_period = bitfield_field32_write(
+      entropy_period, KMAC_ENTROPY_PERIOD_WAIT_TIMER_FIELD, kKmacWaitTimerMax);
+  abs_mmio_write32(kKmacBaseAddr + KMAC_ENTROPY_PERIOD_REG_OFFSET,
+                   entropy_period);
+
+  // Configure max for hash threshold
+  uint32_t entropy_hash_threshold =
+      KMAC_ENTROPY_REFRESH_THRESHOLD_SHADOWED_REG_RESVAL;
+  entropy_hash_threshold = bitfield_field32_write(
+      entropy_hash_threshold,
+      KMAC_ENTROPY_REFRESH_THRESHOLD_SHADOWED_THRESHOLD_FIELD,
+      kKmacEntropyThresholdMax);
+  abs_mmio_write32(
+      kKmacBaseAddr + KMAC_ENTROPY_REFRESH_THRESHOLD_SHADOWED_REG_OFFSET,
+      entropy_hash_threshold);
+
+  // Configure CFG
+  uint32_t cfg_reg = KMAC_CFG_SHADOWED_REG_RESVAL;
+  // Little_endian
+  cfg_reg =
+      bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_MSG_ENDIANNESS_BIT, 0);
+  cfg_reg =
+      bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_STATE_ENDIANNESS_BIT, 0);
+
+  // Sideload: off, default key comes from SW
+  cfg_reg = bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_SIDELOAD_BIT, 0);
+
+  // Entropy mode: EDN
+  // cfg_reg = bitfield_field32_write(cfg_reg,
+  // KMAC_CFG_SHADOWED_ENTROPY_MODE_FIELD,
+  // KMAC_CFG_SHADOWED_ENTROPY_MODE_VALUE_EDN_MODE);
+  cfg_reg =
+      bitfield_field32_write(cfg_reg, KMAC_CFG_SHADOWED_ENTROPY_MODE_FIELD,
+                             KMAC_CFG_SHADOWED_ENTROPY_MODE_VALUE_SW_MODE);
+
+  // Use quality randomness for message blocks too
+  cfg_reg = bitfield_bit32_write(cfg_reg,
+                                 KMAC_CFG_SHADOWED_ENTROPY_FAST_PROCESS_BIT, 0);
+  // Do not remask message blocks
+  cfg_reg = bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_MSG_MASK_BIT, 0);
+
+  // Mark entropy source as ready
+  cfg_reg =
+      bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_ENTROPY_READY_BIT, 0);
+  // Err not processed
+  cfg_reg =
+      bitfield_bit32_write(cfg_reg, KMAC_CFG_SHADOWED_ERR_PROCESSED_BIT, 0);
+  // Unsupported modes: disabled
+  cfg_reg = bitfield_bit32_write(
+      cfg_reg, KMAC_CFG_SHADOWED_EN_UNSUPPORTED_MODESTRENGTH_BIT, 0);
+
+  abs_mmio_write32_shadowed(kKmacBaseAddr + KMAC_CFG_SHADOWED_REG_OFFSET,
+                            cfg_reg);
+
+  return kKmacOk;
+}
 
 /**
  * Wait until given status bit is set.
  *
  * Loops until the `bit_position` of status register reaches the value
  * `bit_value`.
- * @params bit_position The bit position in the status register.
- * @oarams bit_value Whether it should wait for 0 or 1.
+ * @param bit_position The bit position in the status register.
+ * @param bit_value Whether it should wait for 0 or 1.
  */
 static kmac_error_t wait_status_bit(uint32_t bit_position, uint32_t bit_value) {
```

Comment:
nit: if `bit_value` can only be 0 or 1, maybe a boolean is clearer?
```suggestion
static kmac_error_t wait_status_bit(uint32_t bit_position, bool bit_value) {
```